packages/axelar-wasm-std/src/voting.rs

```diff
@@ -86,16 +169,208 @@ impl WeightedPoll {
             poll_size,
             votes: vec![Uint256::zero(); poll_size],
             status: PollStatus::InProgress,
+            voted: HashSet::new(),
         }
     }
 }
 
 impl Poll for WeightedPoll {
-    fn tally() -> Result<PollResult, Error> {
-        todo!()
+    fn tally(&mut self, block_height: u64) -> Result<PollResult, Error> {
+        if block_height < self.expires_at && self.voted.len() != self.snapshot.participants.len() {
```

Comment:
Why is there that second check?

Reply:
can tally early if all participants voted

Reply:
ah gotcha.. Can you change the `!=` to `<` instead then? I know it's the same, but confusing.